packages/providers/audit-logs-local/src/index.ts

```diff
@@ -1,43 +1,53 @@
-'use strict';
-
-const auditLogContentType = require('./content-types/audit-log');
-
-const provider = {
-  async register({ strapi }) {
+import type { Strapi } from '@strapi/strapi';
+import { schema as auditLogContentType } from './content-types/audit-log';
+
+interface Event {
+  action: string;
+  date: Date;
+  userId: string | number;
+  payload: Record<string, unknown>;
+}
+
+interface Log extends Omit<Event, 'userId'> {
+  user: string | number;
+}
+
+export = {
+  async register({ strapi }: { strapi: Strapi }) {
     const contentTypes = strapi.container.get('content-types');
     if (!contentTypes.keys().includes('admin::audit-log')) {
       strapi.container.get('content-types').add('admin::', { 'audit-log': auditLogContentType });
     }
 
     // Return the provider object
     return {
-      async saveEvent(event) {
-        // Rewrite userId key to user
-        const auditLog = { ...event, user: event.userId };
-        delete auditLog.userId;
+      async saveEvent(event: Event) {
+        const { userId, ...rest } = event;
+
+        const auditLog: Log = { ...rest, user: userId };
```

Comment:
I find this very weird, why do you have to cast `Log`? isn't `userId` at this point still `number | string`?

Reply:
a log doesn't have a userId I'm not sure I understand your question ?

Reply:
Sorry I think I misunderstood at the time 🤔 dw about it